docs/source/virtual_rainforest/core/data.md

```diff
@@ -0,0 +1,221 @@
+initial forcing data for the model - things like air temperature, elevation and
+photosynthetically active radiation - but also  includes the storage of internal
+variables calculated by the various models running within the simulation. The data
+handling for simulations is managed by the {mod}`~virtual_rainforest.core.data` module
+and the {class}`~virtual_rainforest.core.data.Data` class, which provides the data
+loading and storage functions for the Virtual Rainforest. The data system is extendable
+to provide support for different file formats and axis validation (see the [module API
+docs](../../api/core/data.md)) but that is beyond the scope of this document.
+
+A Virtual Rainforest simulation will have one instance of the
+{class}`~virtual_rainforest.core.data.Data` class to provide access to the different
+forcing and internal variables used in the simulation. As they are loaded, all variables
+are validated and then  added to an {class}`xarray.Dataset` object, which provides a
+consistent indexing and data manipulation for the underlying arrays of data.
+
+In many cases, a user will simply provide a configuration file to set up the data that
+will be validated and loaded when a simulation runs, but the main functionality for
+working with data using Python are shown below.
+
+## Validation
+
+One of the main functions of the {mod}`~virtual_rainforest.core.data` module is to
+automatically validate data before it is added to the `Data` instance. Validation is
+applied along a set of **core axes** used in the simulation. For a given core axis:
+
+* The dimension names of a dataset are used to identify if data should be validated on
+  that axis. For example, a dataset with `x` and `y` dimensions will be validated
+  on the `spatial` core axis.
+
+* The axis will have a set of defined validators, which are provided to handles different
+  possible data configurations. For example, there is a specific `spatial` validator
+  used to handle a dataset with `x` and `y` dimensions but no coordinate values.
+
+* When a dataset is checked against a core axis, the validation checks to see that one
+  of those validators applies to the actual configuration of the data, and then runs the
+  specific validation for that configuration.
+
+The validation process is primarily intended to check that the sizes or coordinates of
+the dimensions of provided datasets are congruent with the configuration of a particular
+simulation. Validators may also standardise or subset input datasets to map them onto a
+particular axis configuration.
+
+For more details on the different core axes and the alternative mappings applied by
+validators see the [core axis](axes.md) documentation.
+
+## Creating a `Data` instance
+
+A  {class}`~virtual_rainforest.core.data.Data` instance is created using information
+that provides information on the core configuration of the simulation. At present, this
+is just the spatial grid being used.
```

Comment:
Find this phrasing a little bit clunky, "At present, only a spatial grid can be used to populate a Data instance" or something similar maybe?